atacworks/dl4atac/utils.py

```diff
@@ -440,6 +443,12 @@ def _get_tiling_intervals(sizes, intervalsize):
         chrom = sizes.iloc[i, 0]
         chrend = sizes.iloc[i, 1]
         starts = range(0, chrend - (intervalsize + 1), intervalsize)
+        if chrom_range is not None:
```

Comment:
Let's not embed this within `_get_tiling_intervals`. It serves a different purpose and seems completely disconnected from the original function. I'd suggest making this a separate function that gets called if regions are supplied.

Reply:
done.